pkg/controller/accountclaim/reuse.go

```diff
@@ -44,6 +44,22 @@ func (r *ReconcileAccountClaim) finalizeAccountClaim(reqLogger logr.Logger, acco
 		return err
 	}
 
+	// Get latest version of the account
+	refreshedAccount, err := r.getClaimedAccount(accountClaim.Spec.AccountLink, awsv1alpha1.AccountCrNamespace)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get claimed account")
+		return err
+	}
+
+	reqLogger.Info("Rotating Credentials")
```

Comment:
Can we be more descriptive here and in the error? Reading the logs it would be confusing to know where the rotation is happening and also where the status update failed.